src/index.ts

```diff
@@ -229,6 +229,43 @@ function encodeCashAddr(data: Buffer): string {
   }
 }
 
+function decodeNearAddr(data: string): Buffer {
+  const regex = /(^(([a-z\d]+[\-_])*[a-z\d]+\.)*([a-z\d]+[\-_])*[a-z\d]+$)/g;
+
+  if(!regex.test(data)) {
+    throw Error('Invalid address string');
+  }
+
+  let ndata  = data;
+  if(data.startsWith('0x')) {
+    ndata = data.substr(2);
+  }
+  
+  if(data.length >= 2 || data.length <= 64) {  
+    if((data.endsWith('.near') && data.length > 5)) {
```

Comment:
hm, the account id "blah.com" is also valid.
Top Level Accounts are coming at some point in 2021 -- https://nomicon.io/DataStructures/Account.html#top-level-accounts

Reply:
It can accept ".com".

Reply:
I think @ilblackdragon meant that. `.com` is also acceptable name, not just `.near` and `.com`. In their doc,  `over.9000` and `a.bro` are also added as "examples" of acceptable domains. I don't think you need suffix check at all.